frontend/src/utils/keycloakHook.js

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useRef } from 'react';
+import { useKeycloak } from '@react-keycloak/web';
+
+const useCustomKeycloak = ({ customNonce } = {}) => {
+  let nonce = customNonce;
+  if (!nonce) {
+    customNonce = createUUID();
+  }
+  const { keycloak, initialized } = useKeycloak();
+  const origCreateLoginUrl = keycloak.createLoginUrl;
+  keycloak.createLoginUrl = (options) => {
+    let url = origCreateLoginUrl(options);
+    if (!url.includes('&nonce=')) {
```

Comment:
would this always be false if `useNonce = false` ?

Reply:
I think if useNonce = false, url.includes('&nonce=') would be false the first time this hook is called. So !url.includes('&nonce) would be true the first time, and false subsequent times. I noticed that if you don't have this conditional there, there would be multiple nonce parameters in the call to keycloak server, and it would give you an error.